core/integration/settings.py

```diff
@@ -139,6 +149,11 @@ class ConfigurationFormItemType(Enum):
     SELECT = "select"
     LIST = "list"
     MENU = "menu"
+    NUMBER = "number"
+
+    @classmethod
+    def options_from_enum(cls, enum_: Type[Enum]) -> Dict[Enum | str, str]:
```

Comment:
The return type is actually Dict[str, str] but due to type invariance of the `ConfigurationFormItem.options` attribute we must match the typing.